testing/checkFilterFits.py

```diff
@@ -0,0 +1,288 @@
+#                                                                             #
+# THE SOFTWARE IS PROVIDED 'AS IS', WITHOUT WARRANTY OF ANY KIND, EXPRESS OR  #
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,    #
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE #
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER      #
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING     #
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER         #
+# DEALINGS IN THE SOFTWARE.                                                   #
+#                                                                             #
+###############################################################################
+
+"""
+This script checks that the filter fits stored at RMG-database/input/FilterArrheniusFits are up to date with the current
+database
+"""
+import numpy as np
+import os.path
+import yaml
+import operator
+import math
+
+from rmgpy.data.rmg import RMGDatabase
+from rmgpy.rmgobject import RMGObject
+from rmgpy.quantity import ScalarQuantity
+from rmgpy.kinetics.arrhenius import Arrhenius
+from rmgpy import settings
+from rmgpy.rmgobject import recursive_make_object
+from rmgpy.thermo.thermoengine import submit
+
+import nose.tools
+
+
+class TestArrheniusFilterFits():  # cannot inherit from unittest.TestCase if we want to use nose test generators
+    """
+    Contains unit tests for the database for rigorous error checking.
+    """
+
+    @classmethod
+    def setUpClass(self):
+        """
+        Load the database before running the tests.
+        """
+        databaseDirectory = settings['database.directory']
+        self.database = RMGDatabase()
+        self.database.load(databaseDirectory, thermoLibraries=['primaryThermoLibrary', 'Klippenstein_Glarborg2016',
+                                                               'BurkeH2O2', 'thermo_DFT_CCSDTF12_BAC', 'CBS_QB3_1dHR',
+                                                               'DFT_QCI_thermo', 'Narayanaswamy', 'Lai_Hexylbenzene',
+                                                               'SABIC_aromatics', 'vinylCPD_H'],
+                           kineticsFamilies='all')
+
+    def test_Arrhenius_fits(self):
+
+        ## Load file with Arrhenius fits.
+        databaseDirectory = settings['database.directory']
+        filename = 'FilterArrheniusFits.yml'
+        path = os.path.join(databaseDirectory, filename)
+        path_notebook = os.path.join(databaseDirectory, '../scripts/generateFilterArrheniusFits.ipynb')
+
+        obj = ArrheniusRMGObject()
+
+        class_dictionary = {'ScalarQuantity': ScalarQuantity,
+                            'Arrhenius': Arrhenius,
+                            'ArrheniusRMGObject': ArrheniusRMGObject,
+                            }
+
+        self.load_yaml(obj, path, class_dictionary)
```

Comment:
Change this line to `obj.load_yaml(path, class_dictionary)`. This will allow you to get rid of the extra function in line 169 that is being flagged by codacy.

Reply:
I removed `load_yaml` and replaced it with the method we already have in `rmgpy/data/kinetics/database.py`. I also did some more clean-up and replaced `ArrheniusRMGObject` by loading this class from `rmgpy/data/kinetics/database.py`.